api/user.js

```diff
@@ -0,0 +1,28 @@
+const firebase = require('firebase');
+const db = firebase.firestore();
+
+function addShareToken(userEmail, shareToken) {
+  const userRef = db.collection('users').doc(userEmail);
+  userRef.update({
+    shareTokens: firebase.firestore.FieldValue.arrayUnion(shareToken)
+  }).then(() => {
+    console.log('success!')
+  })
+  .catch(err => {
+    console.error(err);
+    throw err;
+  });
+}
+
+function getShareTokens(userEmail) {
+  db.collection('users').doc(userEmail).get()
+  .then((user) => {
+    return user.data().shareTokens;
```

Comment:
Need to ensure this always returns an array, even if empty (or else update the map you use when iterating over the share tokens to handle undefined cases).